pkg/image/resolve.go

```diff
@@ -96,39 +92,87 @@ func GetImages(systemChartPath, chartPath string, externalImages map[string][]st
 	return imagesList, imagesAndSourcesList, nil
 }
 
-func setImages(source string, imagesFromArgs []string, imagesSet map[string]map[string]bool) {
+func AddImagesToImageListConfigMap(cm *v1.ConfigMap, rancherVersion, systemChartsPath string) (err error) {
+	var windowsImages []string
+	var linuxImages []string
+	exportConfig := ExportConfig{
+		SystemChartsPath: systemChartsPath,
+		OsType:           Windows,
+		RancherVersion:   rancherVersion,
+	}
+	windowsImages, _, err = GetImages(exportConfig, nil, []string{}, nil)
+	if err != nil {
+		return
+	}
+	exportConfig.OsType = Linux
+	linuxImages, _, err = GetImages(exportConfig, nil, []string{}, nil)
+	if err != nil {
+		return
+	}
+
+	cm.Data = make(map[string]string, 2)
+	cm.Data[osTypeImageListName[Windows]] = strings.Join(windowsImages, imageListDelimiter)
+	cm.Data[osTypeImageListName[Linux]] = strings.Join(linuxImages, imageListDelimiter)
+
+	return
+}
+
+func ParseCatalogImageListConfigMap(cm *v1.ConfigMap) (windowsImageList, linuxImageList []string) {
+	windowsImageList = strings.Split(cm.Data[osTypeImageListName[Windows]], imageListDelimiter)
+	linuxImageList = strings.Split(cm.Data[osTypeImageListName[Linux]], imageListDelimiter)
+
+	return
```

Comment:
I will make this change in the next commit. Same as comment below, it shows up as if I added it because I re-organized resolve.go to place the most important functions at the top. I agree with the ridiculous return thing 😆